File: lib/common/coordinate.dart
import '../model/amap_base_model.dart';

class Coordinate extends AMapBaseModel {
  double latitude;
  double longitude;

  Coordinate(this.latitude, this.longitude);

  Coordinate.fromJson(Map<String, dynamic> json)
      : latitude = json['latitude'],
        longitude = json['longitude'];

  @override
  Map<String, dynamic> toJson() {
    return {
      'latitude': latitude,
      'longitude': longitude,
    };
  }
}

File: android/src/main/java/amap/com/example/flutter_amap_plugin/Map/Coordinate.java
package amap.com.example.flutter_amap_plugin.Map;


import com.amap.api.maps.model.CameraPosition;
import com.amap.api.maps.model.LatLng;

public class Coordinate {
    public double latitude;
    public double longitude;

    public Coordinate() {
        //default
    }

    public Coordinate(double lat, double lng) {
        this.latitude = lat;
        this.longitude = lng;
    }

    CameraPosition toCameraPosition() {
        return new CameraPosition(new LatLng(latitude,longitude), 10, 0, 0);
    }

    LatLng toLatLng() {
        return new LatLng(latitude, longitude);
    }
}
